cmd/proxy/proxy.go

```diff
@@ -0,0 +1,426 @@
+
+	return &cli.Command{
+		Name:          "proxy",
+		Description:   `Proxy command.`,
+		Configuration: tConfig,
+		Run: func(_ []string) error {
+			return runCmd(&tConfig.Configuration)
+		},
+		Resources: loaders,
+	}
+}
+
+func runCmd(staticConfiguration *static.Configuration) error {
+	configureLogging(staticConfiguration)
+
+	http.DefaultTransport.(*http.Transport).Proxy = http.ProxyFromEnvironment
+
+	if err := roundrobin.SetDefaultWeight(0); err != nil {
+		log.WithoutContext().Errorf("Could not set round robin default weight: %v", err)
+	}
+
+	staticConfiguration.SetEffectiveConfiguration()
+
+	if err := staticConfiguration.ValidateConfiguration(); err != nil {
+		return err
+	}
+
+	log.WithoutContext().Infof("Traefik version %s built on %s", version.Version, version.BuildDate)
+
+	jsonConf, err := json.Marshal(staticConfiguration)
+	if err != nil {
+		log.WithoutContext().Errorf("Could not marshal static configuration: %v", err)
+		log.WithoutContext().Debugf("Static configuration loaded [struct] %#v", staticConfiguration)
+	} else {
+		log.WithoutContext().Debugf("Static configuration loaded %s", string(jsonConf))
+	}
+
+	stats(staticConfiguration)
+
+	svr, err := setupServer(staticConfiguration)
+	if err != nil {
+		return err
+	}
+
+	ctx := cmd.ContextWithSignal(context.Background())
+
+	if staticConfiguration.Ping != nil {
+		staticConfiguration.Ping.WithContext(ctx)
+	}
+
+	svr.Start(ctx)
+	defer svr.Close()
+
+	sent, err := daemon.SdNotify(false, "READY=1")
+	if !sent && err != nil {
+		log.WithoutContext().Errorf("Failed to notify: %v", err)
+	}
+
+	t, err := daemon.SdWatchdogEnabled(false)
+	if err != nil {
+		log.WithoutContext().Errorf("Could not enable Watchdog: %v", err)
+	} else if t != 0 {
+		// Send a ping each half time given
+		t /= 2
+		log.WithoutContext().Infof("Watchdog activated with timer duration %s", t)
+		safe.Go(func() {
+			tick := time.NewTicker(t)
+			for range tick.C {
+				resp, errHealthCheck := healthcheck.Do(*staticConfiguration)
+				if resp != nil {
+					_ = resp.Body.Close()
+				}
+
+				if staticConfiguration.Ping == nil || errHealthCheck == nil {
+					if ok, _ := daemon.SdNotify(false, "WATCHDOG=1"); !ok {
+						log.WithoutContext().Error("Fail to tick watchdog")
+					}
+				} else {
+					log.WithoutContext().Error(errHealthCheck)
+				}
+			}
+		})
+	}
+
+	svr.Wait()
+	log.WithoutContext().Info("Shutting down")
+
+	return nil
+}
+
+func setupServer(staticConfiguration *static.Configuration) (*server.Server, error) {
+	providerAggregator := aggregator.NewProviderAggregator(*staticConfiguration.Providers)
+
+	// adds internal provider
+	err := providerAggregator.AddProvider(traefik.New(*staticConfiguration))
+	if err != nil {
+		return nil, err
+	}
+
+	tlsManager := traefiktls.NewManager()
+
+	acmeProviders := initACMEProvider(staticConfiguration, &providerAggregator, tlsManager)
+
+	serverEntryPointsTCP, err := server.NewTCPEntryPoints(staticConfiguration.EntryPoints)
```

Comment:
I didn't thought about it when I fist saw your PR, but this is going to open the door to dynamic allocation of entrypoints! Currently, when you initialize Maesh you have to provide the maximum number of HTTP and TCP entrypoints, as your cluster grows, there will be less and less entrypoint available, until you have no entrypoint left. This is especially annoying for TCP as you needs as many TCP entrypoints as you have TCP Services in your cluster.
With this new Proxy using the Traefik library, we will be able to potentially derive the number entrypoints needed from the dynamic configuration :)